### PR TITLE
Topic/nrt help rewrite

### LINKED ISSUE #3916: NRT synthesis guide's first example is broken
The NRT synthesis guide begins with an attempt to open a file for writing, using a relative path. This is not guaranteed to succeed.

IMO this example should be moved into an optional section explaining the binary file format. Users do not need to master this example to use NRT. But, because the example is first, users could (and do) take it to be the optimal method to write a score file. As such, it is misleading to put it here. A new example should be written that demonstrates Score.

### PR DESCRIPTION
**Pull request by jamshark70**
Purpose and Motivation
----------------------

sc-users discussion turned up that the current NRT guide document is not really informative, maybe even misleading, for users new to NRT processing.

This is a draft at an almost complete rewrite.

Fixes #3916. Replaces #3944 (closed because of git usage errors on my part).

Types of changes
----------------

- Documentation (non-code change which corrects or adds documentation for existing features)

Checklist
---------

- [x] Updated documentation, if necessary
- [x] This PR is ready for review (maybe?)

Remaining Work
-------------

~~I have not extensively proofread.~~ I've addressed review comments by Brian and Marcin made under #3944.

Possibly some of the contents might be reordered.

I encourage revisions from other authors. Brian suggested that I push the branch directly to supercollider/supercollider so that other authors could contribute.

### CODE REVIEW

File: HelpSource/Guides/Non-Realtime-Synthesis.schelp [hunk 1]
```diff
@@ -3,112 +3,420 @@ summary:: Non-realtime synthesis with binary files of OSC commands
...
+::
 
-First create an OSC command file (i.e. a score)
 code::
-f = File("Cmds.osc","w");
+(
+var server = Server(\nrt,
+	options: ServerOptions.new
+	.numOutputBusChannels_(2)
+	.numInputBusChannels_(2)
```
[the remark this answers is not in the document]
Comment (jamshark70) [reply HelpSource/Guides/Non-Realtime-Synthesis.schelp]: I agree that setting the sample rate explicitly should be included as a best practice. I don't agree that setting number of channels should be omitted as a best practice. I think we should model ideal behavior (set basically everything explicitly) and then users may omit specific settings at their own risk.

Btw I'm not able to put a lot of time into revisions now, maybe 3 weeks later I can do it. A show on Dec. 16 takes priority at the moment. (Writing this on the phone.)

File: HelpSource/Guides/Non-Realtime-Synthesis.schelp [hunk 2]
```diff
@@ -3,112 +3,420 @@ summary:: Non-realtime synthesis with binary files of OSC commands
...
-	)
+SynthDef(\veryLarge, {
+	// ... very large UGen graph...
 }).writeDefFile;
 )
+
+(
+x = Score([
+	// Do NOT put a \d_recv message for \veryLarge here!
+	// Allow the NRT server to read it from the disk file.
```
[the remark this answers is not in the document]
Comment (jamshark70) [reply HelpSource/Guides/Non-Realtime-Synthesis.schelp]: Sure, good point.

File: HelpSource/Guides/Non-Realtime-Synthesis.schelp [hunk 3]
```diff
@@ -3,112 +3,420 @@ summary:: Non-realtime synthesis with binary files of OSC commands
...
+	\dur, Pexprand(0.8, 1.25, inf) * Pgeom(0.01, 1.0143978590819, 400),
+	\legato, 3,
+).asScore(120);
+
+x.recordNRT(
+	outputFilePath: "~/nrt.aiff".standardizePath,
+	sampleRate: 44100,
+	headerFormat: "AIFF",
+	sampleFormat: "int16",
+	options: server.options,
```
[the remark this answers is not in the document]
Comment (jamshark70) [reply HelpSource/Guides/Non-Realtime-Synthesis.schelp]: Ok, I'll review this example later.

[the remark this answers is not in the document]
Comment (patrickdupuis) [reply HelpSource/Guides/Non-Realtime-Synthesis.schelp]: This seems to work. There's no need for `server.options`, and we must send the `\default` SynthDef to the server.
```
(
x = Pbind(
	\freq, Pexprand(200, 800, inf),
	\dur, Pexprand(0.8, 1.25, inf) * Pgeom(0.01, 1.0143978590819, 400),
	\legato, 3,
).asScore(10);

x.add([0.0, [\d_recv, SynthDescLib.global[\default].def.asBytes]]);
x.sort;

x.recordNRT(
	outputFilePath: "~/nrt.aiff".standardizePath,
	sampleRate: 44100,
	headerFormat: "AIFF",
	sampleFormat: "int16",
	duration: 10
)
)
```

**Comment by patrickdupuis** [comment]
I'd like this to get into `3.10.2` if possible. The only thing that needs changing is the broken Pattern example. See my comment for a proposed fix.

File: HelpSource/Guides/Non-Realtime-Synthesis.schelp (hunk 3 above)
[the remark this answers is not in the document]
Comment (patrickdupuis) [reply HelpSource/Guides/Non-Realtime-Synthesis.schelp]: I'm ok with providing a `ServerOptions` in this example. To do so, you'd have to fidure out a way to refer to the Server that was created with `.asScore`. I wasn't able to figure out how to do that.

**Pull request merged**